doc/source/commands/runprogram.rst

```diff
@@ -0,0 +1,226 @@
+.. _running:
+
+Running Programs
+================
+
+.. contents:: Contents
+    :local:
+    :depth: 2
+
+KerboScript supports running executing code saved in files on the local or
+archive disks. The terms "program" and "script" are generally used
+interchangably when describing this behavior. Programs can either be run using
+one of the :ref:`runfunctions` or the :ref:`run`.  Supported file types are:
+
+    Raw Text KerboScript files
+        These files contain KerboScript text commands, as if you had typed them
+        directly into the terminal.  They are easily read, written, and edited
+        by humans.  These files traditionally have a ``".ks"`` extension,
+        however it is not required.
+    Compiled Machine Language Files
+        These files contain encoded and compressed machine language opcodes.
+        They are not formatted to allow humans to read, write, or edit the
+        files.  These files traditionally have a ``".ksm"`` extension, however
+        it is not required.
+
+.. seealso::
+    :ref:`Compiling`
+        An explaination of how scripts are compiled, as well as an explaination
+        of how :key:`COMPILE` stores opcodes.
+    :ref:`Filename Warning<filename_case_warning>`
+        Warning detailing issues with case sensitive host file systems.
+
+.. note::
+    If you attempt to run the same program twice from within another script,
+    the previously compiled version will be executed without attempting to
+    recompile even if the original source script has been modified.  When a
+    program has finished executing and returns to the main terminal input, the
+    memory containing the programs is released.  This means that every time
+    that a script file is run from the terminal it is recompiled, even if the script
+    file has not changed.
```

Comment:
I think this phrase:

``When a program has finished executing and returns``

needs to be prefaced by some sort of introductory phrase along the lines of "however", or "on the other hand", or "except when"...  This makes it more clear that we're not trying to clarify the previous situation, but to instead compare and contrast it with what happens in an opposite case.

Without that clause it sounds like a further detail or clarification about the same situation, which is confusing to someone trying to pick apart the sentence and understand what it's saying who doesn't already know how things work.

This seems to contradict itself and is confusing:
"This is how it works.  When doing foo, this (contradictory explanation) is how it works".

But this is much clearer:
"This is how it works.  *except when* doing foo, where *instead* this (contradictory explanation) is how it works".